advanced-coroutines-codelab/finished_code/src/main/java/com/example/android/advancedcoroutines/PlantListViewModel.kt

```diff
@@ -105,16 +98,15 @@ class PlantListViewModel internal constructor(
     init {
         clearGrowZoneNumber()
 
-        growZoneChannel.asFlow()
-            .mapLatest { growZone ->
+        growZoneFlow.mapLatest { growZone ->
                 _spinner.value = true
                 if (growZone == NoGrowZone) {
                     plantRepository.tryUpdateRecentPlantsCache()
                 } else {
                     plantRepository.tryUpdateRecentPlantsForGrowZoneCache(growZone)
                 }
             }
-            .onCompletion {  _spinner.value = false }
+            .onEach { _spinner.value = false }
```

Comment:
Is this change intentional? Why did we have `onCompletion` before?

Reply:
onCompletion was probably a mistake. Since the spinner is set to true on every emit, it should also be set to false each time after loading is complete.